src/cavendish_particle_tracks/_widget.py

```diff
@@ -296,6 +304,35 @@ def _on_click_stereoshift(self) -> StereoshiftDialog:
         dlg.move(point)
         return dlg
 
+    def _on_click_load_data(self) -> None:
+        """When the 'Load data' button is clicked, a dialog opens to select the folder where the data is.
+        The data in the folder is loaded as a stack of images, and the stack is named according to the option selected.
+        """
+        if self.load.currentIndex() < 1:
+            return
+
+        self._test_file_dialog = QFileDialog(self)
+        try:
+            self._test_file_dialog.setFileMode(QFileDialog.Directory)
+
+            folder_name = self._test_file_dialog.getExistingDirectory(
+                self,
+                "Open folder for " + str(self.load.currentText()),
+                "./",
+                QFileDialog.DontUseNativeDialog
+                | QFileDialog.ShowDirsOnly
+                | QFileDialog.DontResolveSymlinks
+                | QFileDialog.HideNameFilterDetails,
+            )
+            if folder_name:
```

Comment:
Needed to handle the `Cancel` button in the dialog. 
Is there a better way to do this?